Reading_a_JWST_ASDF_file.ipynb

```diff
@@ -0,0 +1,366 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Searching for Content"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# To show only where array data is located\n",
+    "af.search(type='NDArrayType')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Find any WCS-related items\n",
+    "sr = af.search('wcs')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sr"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sr.paths"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(af.tree['meta']['wcs'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Using the WCS\n",
+    "\n",
+    "This is one of those tags that tells the Python ASDF library to convert this node and all below it into a GWCS object. By using this, you can convert pixel coordinates into sky coordinates.\n",
+    "\n",
+    "For example, if you wish to see where pixel 100, 200 is on the sky, or the converse, what pixel a sky coordinate lands in:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wcs = af.tree['meta']['wcs']\n",
+    "wcs(100,200) # both ra and dec are in degrees"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wcs.invert(5.9, -72) # return pixel coordinates corresponding to this RA, Dec"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# what is the output coordinate frame and units?\n",
+    "wcs.output_frame"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Command line tools\n",
+    "\n",
+    "There is a command line tool that provides a number of utilities for use with ASDF files. Most relevant for this tutorial are the info. For example, you can get information on an ASDF file by typing at the shell level:\n",
+    "```\n",
+    "asdftool info <filename>\n",
+    "```\n",
+    "\n",
+    "This variant of the asdftool command accepts the following options corresponding to the optional keyword options for the info method:\n",
+    "```\n",
+    "--max-rows MAX_ROWS  maximum number of lines\n",
+    "--max-cols MAX_COLS  maximum length of line\n",
+    "--show-values\n",
+    "--no-show-values\n",
+    "```\n",
+    "\n",
+    "For example:\n",
+    "\n",
+    "```\n",
+    "asdftool info --max-rows 100 myfile.asdf\n",
```

Comment:
We could demonstrate this in a code cell like this:

```
!asdftool info {filepath}
```

Reply:
Well, I didn't know about that trick.  On the other hand, it doesn't work since it apparently loads the file through jwst instead.